src/OSVRTrackedDevice.cpp

```diff
@@ -161,23 +162,127 @@ void OSVRTrackedDevice::DebugRequest(const char* request, char* response_buffer,
     // make use of (from vrtypes.h) static const uint32_t k_unMaxDriverDebugResponseSize = 32768;
 }
 
+#ifdef _WINDOWS
+// Copyright Razer LLC 2016
+// Note: Need a general cross platform solution. This may be something that needs to be subsumed into the server functionality
+bool OSVRTrackedDevice::findHMDMonitor(const char *HMDName, DXGI_OUTPUT_DESC *pOutputDesc){
+	IDXGIFactory * pFactory;
+	HRESULT hr = CreateDXGIFactory(__uuidof(IDXGIFactory), (void**)(&pFactory));
+
+	IDXGIAdapter *pAdapter;
+	IDXGIOutput *pOutput;
+
+	bool found = false;
+
+	DISPLAY_DEVICE adapterDevice, monitorDevice;
+
+	// Setup adapter data structure
+	memset(&adapterDevice, 0, sizeof(DISPLAY_DEVICE));
+	adapterDevice.cb = sizeof(adapterDevice);
+
+	// setup up monitor data structure
+	memset(&monitorDevice, 0, sizeof(DISPLAY_DEVICE));
+	monitorDevice.cb = sizeof(monitorDevice);
+
+	int i = 0;
+	int adapterIndex = 0;
+	int monitorIndex = 0;
+	char deviceString[256];
+
+	while (EnumDisplayDevices(NULL, i, &adapterDevice, 0))
+	{
+
+#ifdef VERBOSE_LOGGING
+		{const std::string msg = "OSVRTrackedDevice::findHMDMonitor(" + std::string(HMDName) + ",...): Adapter Name: " + adapterDevice.DeviceName +
+			" Adapter String: " + adapterDevice.DeviceString + "\n";
+		logger_->Log(msg.c_str()); }
+#endif
+		if (EnumDisplayDevices(adapterDevice.DeviceName, 0, &monitorDevice, 0)) // 
+		{
+
+#ifdef VERBOSE_LOGGING
+			{const std::string msg = "OSVRTrackedDevice::findHMDMonitor(" + std::string(HMDName) + ",...): Monitor Name: " + monitorDevice.DeviceName + 
+				" Monitor String: " + monitorDevice.DeviceString + 
+				" Monitor DeviceID: " + monitorDevice.DeviceID + 
+				" Monitor State: " + std::to_string(monitorDevice.StateFlags) + "\n" ;
+			logger_->Log(msg.c_str());}
+#endif
+			sprintf_s(deviceString, "%s", monitorDevice.DeviceString);
+			if (strstr(deviceString, HMDName)){
+				if (pFactory->EnumAdapters(adapterIndex, &pAdapter) != DXGI_ERROR_NOT_FOUND){
+					if (pAdapter->EnumOutputs(monitorIndex, &pOutput) != DXGI_ERROR_NOT_FOUND)
```

Comment:
Or, in what I like to call "the gauntlet", test the opposite of the condition you want and early-out: if you've made it to the end, past all the conditionals, you've run the gauntlet and you must be OK.